network/discovery_e2e_test.go

```diff
@@ -31,8 +34,14 @@ func TestDiscovery_ConnectedPopulatesRoutingTable(t *testing.T) {
 		t.Fatalf("Unable to join peers, %v", joinErr)
 	}
 
-	assert.Equal(t, servers[0].discovery.RoutingTableSize(), 1)
-	assert.Equal(t, servers[1].discovery.RoutingTableSize(), 1)
+	// make sure each routing table has peer
+	if _, err := WaitUntilRoutingTableToBeFilled(ctx, servers[0], 1); err != nil {
+		t.Fatalf("server 0 should add a peer to routing table but didn't, peer=%s", servers[1].host.ID())
```

Comment:
You didn't want to put `require.NoError` instead and in that case we can avoid if.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1745/commits/4e54b84faaea7e7058609da315771e40243255be